back/src/chat/channel/channel.controller.ts

```diff
@@ -56,14 +57,14 @@ export class ChannelController {
 	}
 
 	@Post(':targetChannelId/join')
-	@UseGuards(...AuthenticatedGuard, IsChannelGuard)
+	@UseGuards(...AuthenticatedGuard, IsChannelGuard, IsUserSocketGuard)
 	@UsePipes(ValidationPipe)
 	async join(
 		@Param('targetChannelId', ParseIntPipe) channelId: number,
 		@Req() req: Request,
 		@Body() data: ChannelPasswordDto
 	) {
-		return await this.channelService.join(req.user, channelId, data.password);
+		return await this.channelService.join(req.user, channelId, data.password, data.socketId);
```

Comment:
Be careful, here socketId and password arguments are inverted 👀

According to the [ChannelService](https://github.com/42rave/transcendence/blob/c7dc827a8e15fad83f045455498dab2fc30bca4e/back/src/chat/channel/channel.service.ts#L158), the join method is prototyped as follows:
```ts
async join(user: User, targetChannelId: number, socketId: string, password?: string): Promise<ChannelConnection> {}
```

I suggest you the following modification:
```suggestion
		return await this.channelService.join(req.user, channelId, data.socketId, data.password);
```